storage/innobase/os/os0file.cc

```diff
@@ -1095,13 +1095,11 @@ static ATTRIBUTE_COLD void os_file_log_buffered()
   log_sys.log_maybe_unbuffered= false;
   log_sys.log_buffered= true;
 }
-# endif
 
 /** @return whether the log file may work with unbuffered I/O. */
 static ATTRIBUTE_COLD bool os_file_log_maybe_unbuffered(const struct stat &st)
 {
   MSAN_STAT_WORKAROUND(&st);
-# ifdef __linux__
```

Comment:
Finally, I understood it. The first `#endif` removal makes the function `os_file_log_maybe_unbuffered()` defined only `#ifdef __linux__`. That is acceptable, and I see that the FreeBSD run passed.